### ISSUE TITLE
FlyleafHost.Wpf: Early initialization of the Surface Window and/or creation events of the Windows

### ISSUE THREAD

**Issue by Symbai** [SuRGeoNix/Flyleaf#304]
I'm using WPF in surface mode (no Flyleaf.Controls.WPF). My drag / drop events are not fired when dropping files over FlyLeafHost control but they do fire when dropping files anywhere else on my window.

I do see the drag effect mouse cursor though. Its just I can set a breakpoint on first line in the events and they dont get hit on FlyLeafHost. 

My XAML code (simplified):
```XAML
<Window
    AllowDrop="True"
    PreviewDragEnter="mediaPlayerElement_DragOver"
    PreviewDragOver="mediaPlayerElement_DragOver"
    PreviewDrop="mediaPlayerElement_Drop">
    
	<! -- Some other controls -->
	   <fl:FlyleafHost
                x:Name="FlyLeafHost"
                Grid.Row="1"
                KeyBindings="Surface"
                Player="{Binding Player}" />
				
	<! -- Some other controls -->
</Window>
```
My C# code:
```C#
public MainWindow()
{
	DataContext = new CustomViewModel(this); //enable Flyleaf engine
	InitializeComponent();
	((CustomViewModel)DataContext).MainWindowLoaded(); //Creating flyleaf config etc.
	FlyLeafHost.AllowDrop = true;
	FlyLeafHost.DragEnter += MyWindow_DragOver;
	FlyLeafHost.DragOver += MyWindow_DragOver;
	FlyLeafHost.Drop += MyWindow_Drop;
}

private void MyWindow_Drop(object sender, DragEventArgs e)
{
	if (e.Data.GetDataPresent(DataFormats.FileDrop))
	{
		//Do stuff
	}
}

private void MyWindow_DragOver(object sender, DragEventArgs e)
{
	e.Effects = DragDropEffects.Copy;
}
```

**Comment by SuRGeoNix** [comment]
Please check the wiki to understand how FlyleafHost works for WPF (with the Surface/Overlay windows). You will need to listen to the FlyleafHost.Surface drag drop events for that. Let me know if you need more help.

**Comment by Symbai** [comment]
Sorry, I've checked the wiki but I probably missed the topic about events on WPF. That works but `Surface` property is `null`, even when called in my window `loaded` event. I have to call it delayed which seems a bit odd. Is there an event firing when Flyleaf's `Host_Loaded` (i guess this is where Surface is set) is finished? Or can this set in XAML somehow?

**Comment by SuRGeoNix** [comment]
@Symbai I see your point now, I will review if possible in early initialization of the Surface or at least add events after the creation for both Surface/Overlay.

Also, when I was saying 'Wiki' -I know it's not very detailed yet- I mainly meant to understand that it creates two windows so you mouse it's not over the FlyleafHost control but over the Surface/Overlay windows.

I'm not sure what you meant with the Surface mode and with (no Flyleaf.Controls.WPF). I guess that you don't use an overlay or standalone mode? Standalone mode is when you have a Surface window and the Overlay window is your Main Window. So you don't actually render a FlyleafHost (attached) control that Surface/Overlay need to follow within your Main Window.

**Comment by Symbai** [comment]
> I mainly meant to understand that it creates two windows

<details> 
  <summary>Offtopic </summary>
   This is what I read in another comment. But there Surface wasn't used, instead Player.Controls for events (when using the FLyleaf.Controls.WPF nuget package). But when using this nuget package I have to set window transparency etc. I also have my own controls for things like volume, etc. So I sticked to the code of the MVVM sample and the main difference is that it uses a FlyleafHost control + Surface instead of overlay mode. That's why I tried it on FlyleafHost control first, I didn't thought about checking for `Surface` property. Instead of Wiki, one or two lines about using the events in both examples would probably also work. Or just a comment in code.

Btw: This project is freaking amazing. Opening an 8k x265 movie + jumping around the timeline is almost instantly. The WPF mediaplayer struggles on that. Even the windows 11 player app. I've never seen a C# / WPF video player library performing so good.
</details>
